test/test_images/probe_helper/probe_helper_test.go

```diff
@@ -29,52 +49,346 @@ const (
+	}
+	for {
+		select {
+		case req := <-gotRequest:
+			if req.method == "POST" && req.url == testStorageUploadRequest && strings.Contains(req.body, testStorageCreateBody) {
+				// This request indicates the client's intent to create a new object.
+				finalizeEvent := newCloudEvent(
+					cloudEvent{
+						CeID:      "1234567890",
+						CeSubject: schemasv1.CloudStorageEventSubject("cloudstoragesource-probe-1234567890"),
+						CeType:    schemasv1.CloudStorageObjectFinalizedEventType,
+						CeSource:  schemasv1.CloudStorageEventSource(testStorageBucket),
+					},
+				)
+				if res := c.Send(ctx, *finalizeEvent); !cloudevents.IsACK(res) {
+					logger.Fatalf("Failed to send object finalized CloudEvent from the dummy CloudStorageSource: %v", res)
+				}
+			} else if req.method == "PATCH" && req.url == testStorageRequest && strings.Contains(req.body, testStorageUpdateMetadataBody) {
+				// This request indicates the client's intent to update the object's metadata.
+				updateMetadataEvent := newCloudEvent(
+					cloudEvent{
+						CeID:      "1234567890",
+						CeSubject: schemasv1.CloudStorageEventSubject("cloudstoragesource-probe-1234567890"),
+						CeType:    schemasv1.CloudStorageObjectMetadataUpdatedEventType,
+						CeSource:  schemasv1.CloudStorageEventSource(testStorageBucket),
+					},
+				)
+				if res := c.Send(ctx, *updateMetadataEvent); !cloudevents.IsACK(res) {
+					logger.Fatalf("Failed to send object metadata updated CloudEvent from the dummy CloudStorageSource: %v", res)
+				}
+			} else if req.method == "POST" && req.url == testStorageUploadRequest && strings.Contains(req.body, testStorageArchiveBody) {
+				// This request indicates the client's intent to archive the object.
+				archivedEvent := newCloudEvent(
+					cloudEvent{
+						CeID:      "1234567890",
+						CeSubject: schemasv1.CloudStorageEventSubject("cloudstoragesource-probe-1234567890"),
+						CeType:    schemasv1.CloudStorageObjectArchivedEventType,
+						CeSource:  schemasv1.CloudStorageEventSource(testStorageBucket),
+					},
+				)
+				if res := c.Send(ctx, *archivedEvent); !cloudevents.IsACK(res) {
+					logger.Fatalf("Failed to send object archived CloudEvent from the dummy CloudStorageSource: %v", res)
+				}
+			} else if req.method == "DELETE" && req.url == testStorageGenerationRequest {
+				// This request indicates the client's intent to delete the object.
+				deletedEvent := newCloudEvent(
+					cloudEvent{
+						CeID:      "1234567890",
+						CeSubject: schemasv1.CloudStorageEventSubject("cloudstoragesource-probe-1234567890"),
+						CeType:    schemasv1.CloudStorageObjectDeletedEventType,
+						CeSource:  schemasv1.CloudStorageEventSource(testStorageBucket),
+					},
+				)
+				if res := c.Send(ctx, *deletedEvent); !cloudevents.IsACK(res) {
+					logger.Fatalf("Failed to send object deleted CloudEvent from the dummy CloudStorageSource: %v", res)
+				}
+			}
+		}
+	}
+}
+
+//
+type cloudEvent struct {
```

Comment:
why not using CloudEvent SDK type directly?

Reply:
No good reason, just a mistake! Fixed.